scala/src/main/scala/smile/clustering/k-modes/KModes.scala

```diff
@@ -0,0 +1,92 @@
+package smile.clustering.kmodes
+
+import scala.collection.mutable
+import scala.util.Random
+import smile.math.distance.{Distance, Hamming}
+import smile.clustering.clusteringTypes.BinaryClusteringTypes
+/**
+ * @author Beck Gaël
+ * K-Modes scala implementation. K-Modes is the binary equivalent for K-Means. The mean update for centroids is replace by the mode one which is a majority vote among element of each cluster. This algorithm is Hamming oriented because the computation of the mode is the majority vote exclusively for Hamming distance. 
+ * Link towards linked articles or other implementation :
+ * - http://www.irma-international.org/viewtitle/10828/
+ * - https://www.ijecs.in/index.php/ijecs/article/download/3058/2834/
+ * - https://pypi.python.org/pypi/kmodes/
+ * @param data : the dataset with id
+ * @param k : number of clusters
+ * @param epsilon : distance between ancient modes and update modes under which we consider than the algorithm have converged, if and only if all every mode have converged
+ * @param maxIter : number maximal of iteration
+ * @param metric : the dissimilarity measure used
+ **/
+class KModes(data: Seq[(Int, Array[Int])], var k: Int, var epsilon: Double, var maxIter: Int, var metric: Distance[Array[Int]]) extends BinaryClusteringTypes {
```

Comment:
It should implements `PartitionClustering<int[]>`. `KModesModel` should be merged into this class.